### PR TITLE
[CDAP-19071] Fix Preview Runners when ArtifactLocalizer is Disabled

### PR DESCRIPTION
**Pull request by dli357**
Always use remote for artifact fetching in preview runners regardless of data storage implementation, otherwise we get a bunch of Guice implementation not found errors.

### CODE REVIEW

**Comment by dli357** [comment]
Updated PR to use artifact localizer in all situations. This allows us to abstract away the caching layer (ArtifactLocalizerService) from the fetching implementation (ArtifactLocalizerClient).

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java [hunk 1]
```diff
@@ -91,73 +88,50 @@ public class PreviewRunnerModule extends PrivateModule {
...
+    expose(PluginFinder.class);
+
+    bind(PreferencesFetcher.class).to(RemotePreferencesFetcherInternal.class);
+    expose(PreferencesFetcher.class);
+
+    // Read artifact locations from disk when using artifact localizer due to shared mounted read-only PD.
+    install(new LocalLocationModule());
+    expose(LocationFactory.class);
+
+    // Use remote implementation to fetch preferences from AppFab.
```
Comment (albertshau) [inline cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java:129]: duplicate bindings from above?

**Review by albertshau — APPROVED** [review]
Is there anything we need to worry about here from an upgrade perspective? We're not using additional cluster resources are we?

If there are no concerns, lgtm

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java (hunk 1 above)
Comment (dli357) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java:129]: Can you clarify where the duplicates are? I only see one `LocalLocationModule` in the PR.

**Comment by dli357** [comment]
> Is there anything we need to worry about here from an upgrade perspective? We're not using additional cluster resources are we?
> 
> If there are no concerns, lgtm

Unfortunately this will require additional resources; however, for versions 6.7+, preview will altogether fail with this setting disabled anyways.

File: cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java (hunk 1 above)
Comment (albertshau) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java:129]: line 125

Comment (dli357) [reply cdap-app-fabric/src/main/java/io/cdap/cdap/app/preview/PreviewRunnerModule.java:129]: Fixed, thanks!

**Pull request merged**